pilot/pkg/model/push_context.go

```diff
@@ -259,18 +261,28 @@ func (first *PushRequest) Merge(other *PushRequest) *PushRequest {
 		Reason: append(first.Reason, other.Reason...),
 	}
 
-	// Only merge EdsUpdates when incremental eds push needed.
-	if !merged.Full {
-		merged.EdsUpdates = make(map[string]struct{})
-		// Merge the updates
-		for update := range first.EdsUpdates {
-			merged.EdsUpdates[update] = struct{}{}
+	// Merge the configs updated
+	if len(first.ConfigsUpdated) > 0 && len(other.ConfigsUpdated) > 0 {
+		merged.ConfigsUpdated = make(map[resource.GroupVersionKind]map[string]struct{})
+		for update := range first.ConfigsUpdated {
+			merged.ConfigsUpdated[update] = make(map[string]struct{})
 		}
-		for update := range other.EdsUpdates {
-			merged.EdsUpdates[update] = struct{}{}
+		for update := range other.ConfigsUpdated {
+			merged.ConfigsUpdated[update] = make(map[string]struct{})
+		}
+
+		// Merge the configs updated
+		for kind := range merged.ConfigsUpdated {
+			if len(first.ConfigsUpdated[kind]) > 0 && len(other.ConfigsUpdated[kind]) > 0 {
```

Comment:
This is not right,  no need this. Otherwise you will miss this kind when one push request does not include this kind of update